jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenLayerConfigurations.java

```diff
@@ -91,6 +111,73 @@ static JavaLayerConfigurations getForProject(
     return layerBuilder.build();
   }
 
+  /**
+   * Resolves the source files configuration for a War {@link MavenProject}.
+   *
+   * @param project the {@link MavenProject}
+   * @param extraDirectory path to the directory for the extra files layer
+   * @param appRoot root directory in the image where the app will be placed
+   * @return a {@link JavaLayerConfigurations} for the project
+   * @throws IOException if collecting the project files fails
+   */
+  private static JavaLayerConfigurations getForWarProject(
+      MavenProject project, Path extraDirectory, AbsoluteUnixPath appRoot) throws IOException {
+
+    Path explodedWarPath =
+        Paths.get(project.getBuild().getDirectory()).resolve(project.getBuild().getFinalName());
```

Comment:
Also, I think `getFinalName()` may not work (admittedly in rare cases), because [the exploded WAR directory is configurable with `<webappDirectory>` in the Maven WAR plugin, although the default is `${project.build.directory}/${project.build.finalName}`](http://maven.apache.org/plugins/maven-war-plugin/war-mojo.html). For example,

```xml
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-war-plugin</artifactId>
        <version>3.2.2</version>
        <configuration>
          <webappDirectory>${project.build.directory}/my-webapp</webappDirectory>
        </configuration>
      </plugin>
```

```
$ ls target
classes/  generated-sources/  helloworld-1.war  maven-archiver/  maven-status/  my-webapp/
```

But since we will eventually unzip the WAR file, I' think it's OK for now to use `finalName()`.